src/main/java/org/spongepowered/common/data/processor/data/entity/FlyingDataProcessor.java

```diff
@@ -0,0 +1,123 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.base.Optional;
+import net.minecraft.entity.player.EntityPlayer;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableFlyingData;
+import org.spongepowered.api.data.manipulator.mutable.entity.FlyingData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeFlyingData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+
+public class FlyingDataProcessor extends AbstractSpongeDataProcessor<FlyingData, ImmutableFlyingData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof net.minecraft.entity.Entity;
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public Optional<FlyingData> from(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            final SpongeFlyingData flyingData = new SpongeFlyingData();
+            final boolean flying = ((net.minecraft.entity.Entity) dataHolder).isAirBorne;
+            return Optional.<FlyingData>of(flyingData.setFlying(flying));
+        } else {
+            return Optional.absent();
+        }
+
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataHolder dataHolder, FlyingData manipulator, MergeFunction overlap) {
+        if (supports(dataHolder)) {
+            final FlyingData merged = overlap.merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
```

Comment:
should surround the `overlap` with `checkNotNull(overlap, "MergeFunction cannot be null!")`.